### DISCUSSION TITLE
How can pass a variable to the nzExtra template on Collapse element?

### DISCUSSION THREAD

**Discussion by emanuelbadaceanu** [NG-ZORRO/ng-zorro-antd#8554] [Q&A] [answered]
I'm looking to the possibility to pass the variable (id or json) to the template used in nzExtra on collapse element https://ng.ant.design/components/collapse/en. 
Tried the variant with ngTemplateOutlet and ngTemplateOutletContext in a container but this will display an additional gear icon for settings and not pass the value to the template.

**Comment by ParsaArvanehPA** [comment, marked as answer]
Hi @emanuelbadaceanu,
You haven't shared your code, so I can't say for sure why there is an extra gear icon displayed in your `nzExtra` section; as for the other issue, you have to put you `ng-template` inside the `@for`, and your panels array should have the id you want to show. 
This way you can show an id for each of them. Check this [link](https://stackblitz.com/edit/angular-6vuvmi?file=src%2Fapp%2Fapp.component.ts) for the code sample.

**Comment by emanuelbadaceanu** [comment]
Thank you very much @ParsaArvanehPA ! 
The problem was that I had ngFor on nz-collapse-panel and not outside in a for loop and automatically the template didn't had the details to pass.
